### DISCUSSION TITLE
修改schema之后原有的校验仍然保留

### DISCUSSION THREAD

**Discussion by Sharang-heng** [alibaba/formily#3595] [Q&A 2.x]
onFieldReact('rule.*.expecting', (field, form) => {
        const values = field.query('.compareType').get('value');
        if (values === FilterEnum.SECTION) {
          field.componentProps.schema = SectionDefaultSchema;
        } else {
          field.componentProps.schema = InputSchema;
        }
  });

通过被动联动方式监听切换schema 来显示不同内容，但是切换之后原有的校验仍被保留导致阻塞提交，比如切换成InputSchema，但是提交时调用form.submit还是会对SectionDefaultSchema里面的内容进行校验，请问有什么比较好的处理方式？

**Comment by jiojio-jx** [comment]
需求就是清除嘛
core下有 form 实例有 clearErrors、clearWarnnings、clearSuccesss api

**Comment by coolbob1998** [comment]
schema 不是受控的。需要隐藏组件，建议通过设置 display:none
